dashboard/test/models/user_test.rb

```diff
@@ -142,6 +142,19 @@ class UserTest < ActiveSupport::TestCase
     assert_equal user.user_school_infos.where(end_date: nil).count, 1
   end
 
+  test 'update_school_info with custom school does not update user info when user school info does not include a school_id' do
+    original_school_info = create :school_info
+    user = create :teacher, school_info: original_school_info
+    new_school_info = create :school_info_us_other, school_id: nil
+
+    user.update_school_info(new_school_info)
+    assert_equal original_school_info, user.school_info
+    refute_equal new_school_info, user.school_info
+    assert_not_nil user.school_info_id
+
+    assert_equal user.user_school_infos.count, 1
+  end
```

Comment:
I'm pretty sure this test doesn't cover the new behavior you've introduced.

Revisiting the truth table for the logic you've currently got:

| old has NCES id | new has NCES id | new is complete | Update? | Test |
|:---:|:---:|:---:|:---:| --- |
| F | F | F | **No** | We should be testing this case... |
| F | F | T | Yes | |
| F | T | T | Yes | |
| T | F | F | No | |
| T | F | T | No | ...but I think you're testing this case? |
| T | T | T | Yes | |

Reply:
@islemaster Thanks for the detailed feedback/comments.  The truth tables are great!

The desired behavior is to ensure that a user’s school info is only updated to a complete school info.  You are correct in identifying the case below as the intended change:

| old has NCES id | new has NCES id | new is complete | Update? | Test |
|:---:|:---:|:---:|:---:| --- |
| F | F | F | **No** | We should be testing this case... |